src/Renci.SshNet/Security/Cryptography/BlockCipher.cs

```diff
@@ -154,6 +177,16 @@ public override byte[] Decrypt(byte[] input, int offset, int length)
                 throw new InvalidOperationException("Encryption error.");
             }
 
+            if (_padding is not null)
+            {
+                paddingLength = _padding.PadCount(output);
```

Comment:
I added an Unpad() method on my original commit, maybe that's useful to have?
https://github.com/zybexXL/SSH.NET/compare/feature_AES_CSP...zybexXL:SSH.NET:fix_padding?diff=split#diff-2d0250a9e26828476447e03e6a5f970f5a3e1711b09a17b81231c33e1d5c253eR53